internal/workflow/activities/unarchive.go

```diff
@@ -0,0 +1,178 @@
+package activities
+
+import (
+	"context"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+
+	"github.com/go-logr/logr"
+	"github.com/mholt/archiver/v3"
+
+	"github.com/artefactual-sdps/enduro/internal/fsutil"
+)
+
+const UnarchiveActivityName = "unarchive-activity"
+
+// UnarchiveActivity extracts transfer files from an archive (e.g. zip, tgz).
+type UnarchiveActivity struct {
+	logger logr.Logger
+}
+
+type UnarchiveParams struct {
+	// SourcePath is the path of the transfer to be unarchived.
+	SourcePath string
+
+	// StripTopLevelDir indicates whether the top-level "container" directory of
+	// the archive should be excluded from the destination directory.
+	StripTopLevelDir bool
+}
+
+type UnarchiveResult struct {
+	// DestPath is the path to the extracted archive contents.
+	DestPath string
+
+	// IsDir is true if DestPath is a directory.
+	IsDir bool
+}
+
+func NewUnarchiveActivity(logger logr.Logger) *UnarchiveActivity {
```

Comment:
So many versions of this activity already, it would be nice to be able to share it with other repositories.

Reply:
@jraddaoui are you thinking that we would add an unarchive package to https://github.com/artefactual-labs/gotools/, or create a separate repo, or something else?

Reply:
I'd create a new repository in labs, `temporal-activities` or something like that and add this activity there alongside the one from https://github.com/artefactual-sdps/remove-files-activity. Not sure the best way to organize them there, if we should use different packages for each activity o have them all together in a single package.

Reply:
@jraddaoui I think that's a good plan for the future, but I'm going to leave the unarchive activity in this project for this PR; Moving the activity to a new project is too big a task for me to tackle it here.